auslib/db.py

```diff
@@ -1908,6 +1908,11 @@ def update(self, where, what, changed_by, old_data_version, transaction=None, dr
         current_releases = self.select(where=where, columns=[self.name, self.product, self.read_only], transaction=transaction)
         for current_release in current_releases:
             name = current_release["name"]
+
+            if "read_only" in what and current_release["read_only"] != what["read_only"]:
+                self.change_readonly(where, what["read_only"], changed_by, old_data_version, transaction, dryrun, signoffs)
+                continue
```

Comment:
For readonly update, this flow is called when change will be enacted. So calling the functions that changes the state and ignoring the flow below.